hibernate-core/src/main/java/org/hibernate/resource/transaction/backend/jdbc/internal/JdbcResourceLocalTransactionCoordinatorImpl.java

```diff
@@ -84,8 +84,7 @@ public void explicitJoin() {
 
 	@Override
 	public boolean isJoined() {
-		return physicalTransactionDelegate != null && physicalTransactionDelegate.getStatus() == TransactionStatus.ACTIVE;
-
+		return physicalTransactionDelegate != null && getTransactionDriverControl().isActive( true );
```

Comment:
My JDBC test was in 936fb4a, perhaps it wasn't complete?

That aside, the following code would have failed before this fix:

    em.getTransaction().begin();
    em.getTransaction().setRollbackOnly();
    assertTrue( em.isJoinedTransaction() );

Is the concern here that `isJoinedTransaction` should continue to return false for this case?  My interpretation could be wrong here on what is considered _active_ transaction :).